---
title: "Lab10_Johnbauer"
author: "John Bauer"
date: "2024-03-27"
bibliography: references.bib
output: 
  html_document:
    toc: true
    toc_depth: 4
    Theme: yeti
    Highlight: haddock
    code_folding: hide
---

```{r}
library(tidyverse)
# library(plotly)
library(knitr)
library(DT)

```
# Exercise 1

## NEON Site: Toolik Field Station

![Toolik Field Station [@noauthor_photo_nodate]](/cloud/project/Toolik_Field_Station.png)

Located 400 miles north from Fairbanks, Alaska at the foot of the Brooks mountain range, biodiversity at Toolik Field Station is heavily influenced by its harsh winters where temperatures can reach -50‚Å∞F. It is home to a variety of fauna including caribou, loons, voles, and polar bears. Located above the northern tree line, the vegetation in the tundra here mainly consists of birch, willow, sedges and grass. The site contains a large range of soil conditions, including layers of permafrost, created by glacial action [@NEON_collection_2023].   

# Exercise 2

## Gammaproteobacteria

![Uprooted maximum-likelyhood phylogentic tree of gammaproteobactria based on 120 concatenated single copy proteins sequences from 780 reference genomes [@liao_reclassification_2020]. Figure adapted from Figure 2 of Liao et al.](liaoFig2.png)

The class Gammaproteobacteria, under the phylum Proteobacteria, is made up of around 381 genera that thrive in marine, terrestial, and eukaryotic host ecosystems [@liao_reclassification_2020]. Historically, this class has be defined phylogenetically by 16s rRNA sequence homology [@williams_proposal_2013]. Some notable members of this class include *Escherichia coli*, *Vibrio fischeri*, and *Pseudomonas aeruginosa*. This class has great diversity of morphologies with rod, cocci, spirilla, and filaments all represented [@williams_phylogeny_2010]. Additionally, species in class display a variety of trophisms including chemoautotrophs and photoautotrophs [@gao_phylogenomics_2009].

# Exercise 3

```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove = FALSE) %>% 
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
  
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")

NEON_MAGs_Toolik <- NEON_MAGs %>%
  filter(str_detect(`Site`, 'Toolik Field Station'))

NEON_MAGs_Toolik_GPROTEO <- NEON_MAGs_Toolik %>%
  filter(str_detect(`Class`, 'Gammaproteobacteria'))

NEON_MAGs_GPROTEO <- NEON_MAGs %>%
  filter(str_detect(`Class`, 'Gammaproteobacteria'))

```
## Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik)
```

### Taxonomic Breakdown by Phylum of Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik %>%
          count(Phylum))
```

### Taxonomic Breakdown by Class of Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik %>%
          count(Class))
```

### Taxonomic Breakdown by Order of Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik %>%
          count(Order))
```

### Taxonomic Breakdown by Family of Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik %>%
          count(Family))
```

### Taxonomic Breakdown by Genus of Toolik Field Station MAGs
```{r}
datatable(NEON_MAGs_Toolik %>%
          count(Genus))
```

## Taxonomic Breakdown of Toolik Field Station MAGs in Graphical form
```{r, fig.width = 12, fig.height = 12}
NEON_MAGs_Toolik %>% #Phylum into Order
  ggplot(aes(x=fct_rev(fct_infreq(Phylum)), fill=Class))+geom_bar()+coord_flip()+labs(title="Taxonomic Breakdown by Phylum of Toolik Field Station MAGs", x="Phylum", y="n", fill="Class")+theme_classic()

NEON_MAGs_Toolik %>%   #Class into Order
  ggplot(aes(x=Class))+geom_bar(aes(fill=`Order`),position=position_dodge2(width=0.9, preserve="single"))+coord_flip()+facet_wrap(vars(`Phylum`), scales="free_y", ncol=2)+labs(title="Taxonomic Breakdown by Class of Toolik Field Station MAGs", x="Class", y="n", fill="Order")+theme_classic()
  
NEON_MAGs_Toolik %>%   #Order into family
  ggplot(aes(x=Order))+geom_bar(aes(fill=`Family`),position=position_dodge2(width=0.9, preserve="single"))+coord_flip()+facet_wrap(vars(`Class`), scales="free_y", ncol=2)+labs(title="Taxonomic Breakdown by Order of Toolik Field Station MAGs", x="Order", y="n", fill="Family")+theme_classic()

NEON_MAGs_Toolik %>%   #Family into Genus
  ggplot(aes(x=Family))+geom_bar(aes(fill=`Genus`),position=position_dodge2(width=0.9, preserve="single"))+coord_flip()+facet_wrap(vars(`Class`), scales="free_y", ncol=2)+labs(title="Taxonomic Breakdown by Family of Toolik Field Station MAGs", x="Family", y="n", fill="Genus")+theme_classic()

```

## Gammaproteobacteria Site Locations:

```{r, fig.width = 8, fig.height =4}
NEON_MAGs_GPROTEO %>% #Site Locations
  filter(!str_detect(`Site`, "NEON combined assembly")) %>%
  ggplot(aes(x=fct_rev(fct_infreq(`Site`)), fill=`Order`))+geom_bar()+coord_flip()+labs(title="Gammaproteobacteria MAGs Sample Locations", x="Site", y="n", fill="Order")+theme_classic()

datatable(NEON_MAGs_GPROTEO)

```

# Exercise 4
```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  # filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T)) 

NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")

NEON_META_Toolik <- NEON_metagenomes %>%
  filter(str_detect(`Sample Name`,'re-annotation')) %>%
  filter(str_detect(`Site`,'Toolik'))
datatable(NEON_META_Toolik)
```

# References