---
title: "Lab9_JohnBauer"
author: "John Bauer"
date: "2024-03-13"
output: 
  html_document:
    toc: true
    toc_depth: 4
    Theme: yeti
    Highlight: haddock
    code_folding: hide
---
```{r}
library(tidyverse)
library(knitr)
library(ggtree)
library(TDbook) #A Companion Package for the Book "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).
library(ggimage)
library(rphylopic)
library(treeio)
library(tidytree)
library(ape)
library(TreeTools)
library(phytools)
library(ggnewscale)
library(ggtreeExtra)
library(ggstar)
#library(ggeasy) #https://jonocarroll.github.io/ggeasy/
library(DT)

```

# Example
## Example #4 NEON Single Bar Plot

```{r}
#Format data set
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove = FALSE) %>% 
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
  
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 

NEON_MAGs_bact_ind <- NEON_MAGs %>% 
  filter(Domain=="Bacteria") %>% 
  filter(`Assembly Type`=="Individual")
```
```{r}
NEON_MAGs_bact_ind %>% 
  ggplot(aes(x=Phylum))+geom_bar()+coord_flip()

NEON_MAGs_bact_ind %>% 
ggplot(aes(x=fct_infreq(Phylum)))+geom_bar()+coord_flip()+labs(x="Phylum") #order by count

NEON_MAGs_bact_ind %>% 
  count(Phylum) %>% 
ggplot(aes(x = reorder(Phylum, n), y = n)) +
  geom_col(stat = "identity") +
  coord_flip()

NEON_MAGs_bact_ind %>% 
  count(Phylum) %>% 
  ggplot(aes(x=reorder(Phylum, n), y=n), fill=`Phylum`)+geom_col(stat="identity")+coord_flip()+labs(x="Phylum")

```

## Example #5 NEON Multiple Bar plots

```{r, fig.width = 12, fig.height = 12}
# NEON_MAGs_bact_ind %>%
#   ggplot(aes(x=fct_rev(fct_infreq(Phylum)), fill=Site))+geom_bar()+coord_flip()+labs(x="Phylum")
NEON_MAGs_bact_ind_Novel <- NEON_MAGs_bact_ind %>% 
  filter(is.na(Class) | is.na(Order) | is.na(Family) | is.na(Genus) )

NEON_MAGs_bact_ind %>%
  ggplot(aes(x=fct_rev(fct_infreq(Phylum)), fill=Site))+geom_bar(position="dodge")+coord_flip()+labs(title="NEON MAG Distribution by Phylum", x="Phylum", y="MAG Count")+theme_classic()

ggsave(
  "Overall_MAG_Count_per_Phylum.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

# NEON_MAGs_bact_ind %>%
#   ggplot(aes(x=fct_rev(fct_infreq(Phylum)), fill=Site))+geom_bar(position=position_dodge2(width=0.9, preserve= "single"))+coord_flip()

# NEON_MAGs_bact_ind %>%
#   ggplot(aes(x=Phylum))+geom_bar(position=position_dodge2(width=0.9, preserve="single"))+coord_flip()+facet_wrap(vars(Site), scales="free_y", ncol=2)

```

## Example #6 NEON Histogram and Boxplot
```{r, fig.width = 8, fig.height = 8}
NEON_MAGs_bact_ind %>% 
  ggplot(aes(x=`Total Number of Bases`))+geom_histogram(bins=50)

ggsave(
  "Overall_MAG_Count_per_Class.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

NEON_MAGs_bact_ind %>%
  ggplot(aes(x=fct_infreq(Phylum), y=`Total Number of Bases`))+geom_boxplot()+theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1))+labs(x="Phylum")

ggsave(
  "Overall_MAG_Count_per_Class.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

NEON_MAGs_bact_ind %>%
  ggplot(aes(x=fct_infreq(Phylum), y=`Total Number of Bases`))+geom_point()+coord_flip()+labs(x="Phylum")

ggsave(
  "Overall_MAG_Count_per_Class.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

# Exercises

## Exercise #1

What are the overall class MAG counts?
```{r, fig.width = 8, fig.height = 8}
NEON_MAGs_bact_ind %>%
  ggplot(aes(x=fct_rev(fct_infreq(Class)), fill=Class))+geom_bar(show.legend=FALSE)+coord_flip()+labs(title="Overall MAG Count per Class", x="Class", y="MAG Count")+theme_classic()

ggsave(
  "Overall_MAG_Count_per_Class.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #2

What are the MAG counts for each subplot. Color by site ID.
```{r, fig.width = 12, fig.height = 8}
NEON_MAGs_bact_ind %>%
  ggplot(aes(x=fct_rev(fct_infreq(Subplot)), fill=`Site ID`))+geom_bar()+coord_flip()+labs(title="Overall MAG Count per Subplot", x="Subplot ID", y="MAG Count")+theme_classic()

ggsave(
  "Overall_MAG_Count_per_Subplot.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #3

How many novel bacteria were discovered (Show that number of NAs for each taxonomic level)?
```{r, fig.width = 8, fig.height = 8}
NEON_MAGs_bact_ind_Novel <- NEON_MAGs_bact_ind %>% 
  filter(is.na(Class) | is.na(Order) | is.na(Family) | is.na(Genus) ) %>%
  ggplot(aes(x=fct_rev(fct_infreq(Site)), fill=Site))+geom_bar(show.legend=FALSE)+coord_flip()+labs(title="Total Novel Bacteria per Site", x="Site", y="n")+theme_classic()

ggsave(
  "Total_Novel_Bacteria_per_Site.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #4

How many novel bacterial MAGs are high quality vs medium quality?
```{r, fig.width = 4, fig.height = 4}
NEON_MAGs_bact_ind %>% 
  filter(is.na(Class) | is.na(Order) | is.na(Family) | is.na(Genus) ) %>%
  ggplot(aes(x=fct_infreq(`Bin Quality`), fill=`Bin Quality`))+geom_bar()+labs(title="Novel Bacteria MAG Quality", x="Quality", y="MAGs")+theme_classic()

ggsave(
  "Novel_Bacteria_MAG_Quality.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #5
What phyla have novel bacterial genera?
```{r, fig.width = 12, fig.height = 8}
NEON_MAGs_bact_ind %>% 
  filter(is.na(Class) | is.na(Order) | is.na(Family) | is.na(Genus) ) %>%
  ggplot(aes(x=fct_rev(`Phylum`), fill=`Phylum`))+geom_bar(show.legend=FALSE)+coord_flip()+labs(title="Novel Genera per Phylum", x="Phylum", y="n")+theme_classic()

ggsave(
  "Novel_Genera_per_Phylum.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #6

Make a stacked bar plot of the total number of MAGs at each site using Phylum as the fill.
```{r, fig.width = 12, fig.height = 8}
NEON_MAGs_bact_ind %>% 
  ggplot(aes(x=fct_rev(fct_infreq(Site)), fill=Phylum))+geom_bar()+coord_flip()+labs(title="Total Number of MAGs per site", x="Site", y="MAG Count",fill="Phylum")+theme_classic()

ggsave(
  "Total_Number_of_Mags_per_Phylum_at_Each_Site_Stacked.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

```
## Exercise #7

Using facet_wrap make plots of the total number of MAGs at each site for each phylum (e.g. similar to the example above but using the site ID and separating each graph by phylum.)
```{r, fig.width = 12, fig.height = 8}
NEON_MAGs_bact_ind %>%
  ggplot(aes(x=Phylum))+geom_bar(aes(fill=Phylum),position=position_dodge2(width=0.9, preserve="single"),show.legend=FALSE)+coord_flip()+facet_wrap(vars(`Site ID`), scales="free_y", ncol=4)+labs(title="Total Number of Mags per Phylum at Each Site", x="Phylum", y="n")+theme_classic()

ggsave(
  "Total_Number_of_Mags_per_Phylum_at_Each_Site_facet.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Exercise #8

What is the relationship between MAGs genome size and the number of genes? Color by Phylum.
```{r, fig.width = 8, fig.height = 8}
NEON_MAGs_bact_ind %>%
            mutate(`Genome Size (Kbp)`=as.integer(`Total Number of Bases`/1000)) %>%
            ggplot(aes(x=`Gene Count`, y=`Genome Size (Kbp)`, color=Phylum))+geom_point()+labs(title="Gene Count vs Estimated Mag Genome Size", x="Gene", y="Estimated Genome Size (Kbp)")+theme_classic()

ggsave(
  "Gene_Count_vs_Estimated_Mag_Genome_Size.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

```

## Exercise #9

What is the relationship between scaffold count and MAG completeness?
```{r, fig.width = 8, fig.height = 8}
NEON_MAGs_bact_ind %>%
            ggplot(aes(x=`Scaffold Count`, y=`Bin Completeness`, color=Phylum))+geom_point()+labs(title="Scaffold Count vs MAG Completeness", x="Scaffold Count", y="MAG Completeness")+theme_classic()

ggsave(
  "Scaffold_Count_vs_MAG_Completeness.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Lab9",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

