---
title: "Lab8_JohnBauer"
author: "John Bauer"
date: "2024-03-06"
output: 
  html_document:
    toc: true
    toc_depth: 4
    Theme: yeti
    Highlight: haddock
    code_folding: hide
---
```{r}
library(tidyverse)
library(knitr)
library(DT)
```

# Examples

## Example #1
```{r}
  iris_setosa <- iris %>% 
  filter(Species=="setosa") %>% 
  filter(Sepal.Length>5)
kable(iris_setosa)
#data table from object
datatable(iris_setosa)
#data table in one piping
datatable(
  iris %>% 
    filter(Species=="setosa") %>% 
    filter(Sepal.Length>5))
```

## Example #2 NEON
```{r}
NEON_MAGs <- read_csv("~/Bio 676/Lab8/data/GOLD_Study_ID_Gs0161344_NEON.csv")
head(NEON_MAGs)
str(NEON_MAGs)
#remove combined assembly
NEON_MAGs_Ind <- NEON_MAGs %>% 
  filter(`Genome Name`!="NEON combined assembly")#use grave quotes (tilda key)
#count mags
NEON_MAGs_Ind %>%
  count(`Bin Quality`,sort=TRUE)
#kabl table
kable(NEON_MAGs_Ind %>%
        count(`Bin Quality`))
#filter HQ bin quality and data table
datatable(NEON_MAGs_Ind %>%
            filter(`Bin Quality`=="HQ"))
#Select GTDB taxonomy and the MAGs genome size then filter to all MAGs greater than 10,000,000 bases

kable(NEON_MAGs_Ind%>%
        select(c(`GTDB-Tk Taxonomy Lineage`,`Total Number of Bases`)) %>%
        filter(`Total Number of Bases`>10000000))
# datatable filtering on string
datatable(NEON_MAGs_Ind %>%
            filter(str_detect(`GTDB-Tk Taxonomy Lineage`, "Bacteroidota")))
#filter to just Yellowstone
datatable(NEON_MAGs_Ind %>%
            filter(str_detect(`Genome Name`, 'Yellowstone NP')))
#separate taxonomy by ;
NEON_MAGs_Ind_Tax <- NEON_MAGs_Ind %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"),"; ", remove=FALSE)
#count the Phylum contents
datatable(NEON_MAGs_Ind_Tax %>%
            count(Phylum, sort=TRUE))
#separate genome name into columns
NEON_MAGs_Ind_Tax_Sample <- NEON_MAGs_Ind_Tax %>%
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% #this mutates genome name column data to replace the string with ""
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>%
  mutate_at("Sample Name", str_replace, "S-comp-1", "") %>%
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>%
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-",) 
#count MAGs at each site
datatable(NEON_MAGs_Ind_Tax_Sample %>% 
            count(Site, sort = TRUE))
```

# Challenges

## Challenge #1
Use view(iris) to see the whole data table. Subset the table based on a different species than was used in the example. Display the table using DT::datatable
```{r}
#view(iris)
datatable(iris %>%
            filter(Species=="virginica"))
```

## Challenge #2
Display using DT::datatable the NEON MAGs from the individual assemblies that have at least 1 16S rRNA
```{r}
datatable(NEON_MAGs_Ind %>%
            filter(`16s rRNA`>0))

```

## Chalenge #3
Display a table of the MAGs from Lower Teakettle with only the columns for the Genome Name, GTDB-Tk Taxonomy Lineage, and estimated MAG genome size.
```{r}
datatable(NEON_MAGs_Ind_Tax %>%
            filter(str_detect(`Genome Name`, "Lower Teakettle")) %>%
            mutate(`Estimated Genome Size (Kbp)`=as.integer(`Total Number of Bases`/(`Bin Completeness`/100)/1000)) %>%
            select(c("Genome Name","GTDB-Tk Taxonomy Lineage","Estimated Genome Size (Kbp)")))
```

## Challenge #4
Display a table with the Class counts at LBJ National Grasslands
```{r}
datatable(NEON_MAGs_Ind_Tax_Sample %>%
          filter(str_detect(Site, "National Grasslands LBJ")) %>%
          count(Class, sort=TRUE))
```

## Challenge #5
Display a table with the counts for the Phylum Actinobacteriota at each Site
```{r}
datatable(NEON_MAGs_Ind_Tax_Sample %>%
            filter(Phylum=="Actinobacteriota") %>%
            count(Site, sort=TRUE))
```

